### PR TITLE
feat: collector & verifier  high liveness

### PR DESCRIPTION
**Pull request by solargatsby**

### CODE REVIEW

File: offchain-modules/packages/x/src/core.ts [hunk 1]
```diff
@@ -45,6 +58,7 @@ export class ForceBridgeCore {
   private static _ckb: CKB;
   private static _ckbIndexer: CkbIndexer;
   private static _keystore: KeyStore;
+  private static _xChainHandler: XChainHandlers;
```
Comment (homura) [inline offchain-modules/packages/x/src/core.ts:61]: i dont think the handler should be in the `ForceBridgeCore`, `cli, and `rpc` also depend on `ForceBridgeCore`, but they don't need the handler

File: offchain-modules/packages/x/src/handlers/ckb.ts [hunk 2]
```diff
@@ -66,6 +66,15 @@ export class CkbHandler {
     await this.kvDb.set(lastHandleCkbBlockKey, `${blockNumber},${blockHash}`);
   }
 
+  getHandledBlock(): { height: number; hash: string } {
+    return { height: this.lastHandledBlockHeight, hash: this.lastHandledBlockHash };
+  }
+
+  async getTipBlock(): Promise<{ height: number; hash: string }> {
+    const tipHeader = await this.ckb.rpc.getTipHeader();
+    return { height: Number(tipHeader.number), hash: (this.lastHandledBlockHash = tipHeader.hash) };
```
Comment (homura) [inline offchain-modules/packages/x/src/handlers/ckb.ts:75]: an `getTipBlock` should not cause a state change, state changes should be explicit

```ts
const block = handler.getTipBlock();
handler.setLatestHandledBlockHash*(
```

File: offchain-modules/packages/x/src/multisig/multisig-mgr.ts [hunk 3]
```diff
@@ -8,11 +8,13 @@ import { httpRequest } from './client';
 
 const SigErrorTxNotFound = 1003;
 const SigErrorTxUnconfirmed = 1004;
+const SigErrorBlockSyncUncompleted = 1005;
 const SigErrorCodeUnknownError = 9999;
 
 const retryErrorCode = new Map<number, boolean>([
```
Comment (homura) [inline offchain-modules/packages/x/src/multisig/multisig-mgr.ts:19]: there is a `Set` data structure in JS, maybe we can just use a set for managing `RetryErrorCode`, or will the value corresponding to the error code be extended with functions, such as retryTimes, in which case it should be marked with a TODO  here?

File: offchain-modules/packages/x/src/core.ts (hunk 1 above)
Comment (solargatsby) [reply offchain-modules/packages/x/src/core.ts:61]: Handler is a common module,  all of collector, rpc and signer server need handler

File: offchain-modules/packages/x/src/handlers/ckb.ts (hunk 2 above)
Comment (solargatsby) [reply offchain-modules/packages/x/src/handlers/ckb.ts:75]: mistake

**Pull request merged**